src/main/java/teammates/common/datatransfer/questions/FeedbackRubricQuestionDetails.java

```diff
@@ -964,4 +895,103 @@ public void setNumOfRubricSubQuestions(int numOfRubricSubQuestions) {
+         * percentageFrequencyAndAverageValue[subQuestionIndex][choiceIndex]
+         * -> is the percentage choiceIndex is chosen for subQuestionIndex
+         * percentageFrequencyAndAverageValue[subQuestionIndex][numOfRubricChoices]
+         * -> is the average weight of the responses for the given sub-question
+         */
+        float[][] percentageFrequencyAndAverage;
+
+        List<FeedbackResponseAttributes> responses;
+        FeedbackRubricQuestionDetails questionDetails;
+
+        int numOfRubricSubQuestions;
+        int numOfRubricChoices;
+        int responseTotalIndex;
+
+        RubricStatistics(List<FeedbackResponseAttributes> responses, FeedbackRubricQuestionDetails questionDetails) {
+
+            this.responses = responses;
+            this.questionDetails = questionDetails;
+
+            this.numOfRubricSubQuestions = questionDetails.numOfRubricSubQuestions;
+            this.numOfRubricChoices = questionDetails.numOfRubricChoices;
+            this.responseTotalIndex = numOfRubricChoices;
+
+            calculateResponseFrequency();
+            calculatePercentageFrequencyAndAverage();
+        }
+
+        void calculateResponseFrequency() {
+            responseFrequency = new int[numOfRubricSubQuestions][numOfRubricChoices + 1];
+            // Count frequencies
+            for (FeedbackResponseAttributes response : responses) {
+                FeedbackRubricResponseDetails frd = (FeedbackRubricResponseDetails) response.getResponseDetails();
+                for (int i = 0; i < numOfRubricSubQuestions; i++) {
+                    int chosenChoice = frd.getAnswer(i);
+                    if (chosenChoice != -1) {
+                        responseFrequency[i][chosenChoice] += 1;
+                        responseFrequency[i][responseTotalIndex] += 1;
+                    }
+                }
+            }
+        }
+
+        /**
+         * Calculates the percentage frequencies for each choice and average value for each sub-question
+         * Requires responseFrequency to be computed
+         */
+        void calculatePercentageFrequencyAndAverage() {
+            percentageFrequencyAndAverage = new float[numOfRubricSubQuestions][numOfRubricChoices + 1];
+            // calculate percentage frequencies and average value
+            for (int i = 0; i < percentageFrequencyAndAverage.length; i++) {
+                int totalForSubQuestion = responseFrequency[i][responseTotalIndex];
+                //continue to next row if no response for this sub-question
+                if (totalForSubQuestion == 0) {
+                    continue;
+                }
+                // Divide responsesFrequency by totalForSubQuestion to get percentage
+                for (int j = 0; j < numOfRubricChoices; j++) {
+                    percentageFrequencyAndAverage[i][j] = (float) responseFrequency[i][j] / totalForSubQuestion;
+                }
+                // Calculate the average for each sub-question
```

Comment:
To be consistent, all short comments like this should start with a lower case letter?